src/WebJobs.Script/Diagnostics/OpenTelemetry/OpenTelemetryConfigurationExtensions.cs

```diff
@@ -0,0 +1,136 @@
+                .AddFilter<OpenTelemetryLoggerProvider>("Azure.*", _ => !ScriptHost.WorkerOpenTelemetryEnabled)
+                .AddFilter<OpenTelemetryLoggerProvider>("Microsoft.Azure.WebJobs.*", _ => !ScriptHost.WorkerOpenTelemetryEnabled);
+
+            // Azure SDK instrumentation is experimental.
+            AppContext.SetSwitch("Azure.Experimental.EnableActivitySource", true);
+
+            loggingBuilder.Services.AddOpenTelemetry()
+                .ConfigureResource(r => ConfigureResource(r))
+                .WithTracing(b =>
+                {
+                    b.AddSource("Azure.*");
+                    b.AddAspNetCoreInstrumentation();
+                    b.AddHttpClientInstrumentation(o =>
+                    {
+                        o.FilterHttpRequestMessage = _ =>
+                        {
+                            Activity activity = Activity.Current?.Parent;
+                            return (activity == null || !activity.Source.Name.Equals("Azure.Core.Http")) ? true : false;
+                        };
+                    });
+                    if (enableOtlp)
+                    {
+                        b.AddOtlpExporter();
+                    }
+                    if (!string.IsNullOrEmpty(azMonConnectionString))
+                    {
+                        b.AddAzureMonitorTraceExporter(options => options.ConnectionString = azMonConnectionString);
+                        b.AddLiveMetrics(options => options.ConnectionString = azMonConnectionString);
+                    }
+                    b.AddProcessor(ActivitySanitizingProcessor.Instance);
+                    b.AddProcessor(TraceFilterProcessor.Instance);
+                });
+
+            string eventLogLevel = GetConfigurationValue(EnvironmentSettingNames.OpenTelemetryEventListenerLogLevel, context.Configuration);
+            if (!string.IsNullOrEmpty(eventLogLevel))
+            {
+                if (Enum.TryParse(eventLogLevel, ignoreCase: true, out EventLevel level))
+                {
+                    loggingBuilder.Services.AddSingleton<IHostedService>(serviceProvider => new OpenTelemetryEventListenerService(level));
+                }
+                else
+                {
+                    throw new InvalidEnumArgumentException($"Invalid `{EnvironmentSettingNames.OpenTelemetryEventListenerLogLevel}`.");
+                }
+            }
+            else
+            {
+                // Log all warnings and above by default.
+                loggingBuilder.Services.AddSingleton<IHostedService>(serviceProvider => new OpenTelemetryEventListenerService(EventLevel.Warning));
+            }
+
+            static ResourceBuilder ConfigureResource(ResourceBuilder r)
+            {
+                string serviceName = Environment.GetEnvironmentVariable(ResourceAttributeConstants.SiteNameEnvVar) ?? "azureFunctions";
```

Comment:
I am not sure I like the fallback of `azureFunctions`. Is there something better?

Reply:
No longer adding serviceName if the value is null.